### DISCUSSION TITLE
Not able to create dimensions and event locally

### DISCUSSION THREAD

**Discussion by GitVivekHub** [Sunbird-cQube/community#785] [Q&A] [answered]
Hello, Im trying to create dimension and event with given csv files like for example medium dimension with given csv.

But when i try to do so it gives me an error that spec.DimensionGrammar table doesnt exists. I see no such table in the database.

The thing is im connecting postgres database locally with localhost as its hostname. Here im unable to load or see all the tables needed . Due to which whenever i run my application locally and try to ingest an event/dimension the application breaks as my api's cant find those tables

**Comment by RNarmatha** [comment]
Hi @GitVivekHub - Could you please elaborate from the issue . by running   yarn cli ingest commandit will automatically create

**Reply by GitVivekHub** [reply]
The issue is im not able to load or see all the tables required when i connect to my postgres database. Where can i see all the relevant tables.

**Reply by GitVivekHub** [reply]
The thing is im connecting postgres database locally with localhost as its hostname. Here im unable to load or see all the tables needed . Due to which whenever i run my application locally and try to ingest an event/dimension the application breaks as my api's cant find those tables

**Comment by RNarmatha** [comment]
Kindly  follow all the steps like 
1 . External schema(for new programs,VSK, dimension) ingestion using spec api and  
2. data(new programs,VSK, dimension) ingestion to bucket using ingestion  
3. then schedule and run adapter processor group in NiFi 
4. then schedule and run latest code processor group NiFi
Once you  run the latest code processor group then should see all the tables with data

**Reply by GitVivekHub** [reply]
Can you please elaborate on this one with commands and the folder where i can find the code to run the commands. I am new to cQube so just need little more details on it

**Reply by GitVivekHub** [reply, marked as answer]
Also i have csv files present with me that i want to ingest so do i really need to run the adapter service even when i have the csv files with me?

**Reply by GitVivekHub** [reply]
Can you please provide me with example dataset schemas for datasetgrammar table. Dataaset schemas are missing.

https://github.com/Sunbird-cQube/spec-ms/tree/dev/src/JsonSchemas/

In the above link for json schemas i can only  find schemas for dimensions and events but not for dataset schemas

**Comment by RNarmatha** [comment]
Hi @GitVivekHub 
 Kindly follow the cQube [Master](https://docs.google.com/document/d/1wpKSuMNjzwoczznTlhb-fTAc3WdiEc4aWn2a95YIbpU/edit?usp=sharing) Document  for the ingestion

**Reply by GitVivekHub** [reply]
Can you please provide me with example dataset schemas for datasetgrammar table. Dataaset schemas are missing.

https://github.com/Sunbird-cQube/spec-ms/tree/dev/src/JsonSchemas/

In the above link for json schemas i can only find schemas for dimensions and events but not for dataset schemas

**Reply by GitVivekHub** [reply]
Also for spec.EventGrammar and for spec.DimensionGrammar i have created the tables manually. Can you please confirm if this is a correct approach. I did so because my spec/event and spec/dimension api broke as i didnt had these two tables in the database and the code checked these tables before creating event or dimension

**Comment by RNarmatha** [comment]
Hi @GitVivekHub - Manually we can't the Spec and Event.  Kindly follow the steps  which are mention in the document

**Reply by GitVivekHub** [reply]
As per the documentation when im trying to execute the ingestion/new_programs api it gives me error since it cant find spec.EventGrammar and spe.DimenstionGrammar table. So my question is if i dont have to create these tables manually then how can i get them
